pkg/graphql/client.go

```diff
@@ -0,0 +1,89 @@
+package graphql
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+)
+
+type ClientImplConfig struct {
+	HttpClient *http.Client
+	Url        string
+}
+
+type ClientImpl struct {
+	httpClient *http.Client
+	url        string
+}
+
+var _ Client = (*ClientImpl)(nil)
```

Comment:
Check that `ClientImpl` implements the `Client` interface.

This is better than
```go
var _ Client = &ClientImpl{}
```
because it does no memory allocations.

Reply:
I never even saw such code construction in Golang

Reply:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/cm/topologymanager/topology_manager.go#L116

This is for checking during compilation if a type implements an interface. If that type doesn't implement the interface correctly, the compilation would fail

Reply:
If this is not common knowledge, you might want to add a comment, to enlighten the future maintainer :)

Reply:
There's a better worded explanation in Effective Go
https://golang.org/doc/effective_go#blank_implements